routers/api/v1/user/repo.go

```diff
@@ -9,7 +9,7 @@ import (
 // listUserRepos - List the repositories owned by the given user.
 func listUserRepos(ctx *context.APIContext, u *models.User) {
 	userID := u.ID
-	showPrivateRepos := (ctx.User.ID == userID || ctx.User.IsAdmin) && ctx.IsSigned
+	showPrivateRepos := ctx.IsSigned && (ctx.User.ID == userID || ctx.User.IsAdmin)
```

Comment:
ctx.User is also used in `getAccessibleRepos`called from https://github.com/lunny/gitea/blob/07ad1e27dd8fbebb90b53590a958115e7c680130/routers/api/v1/user/repo.go#L50